sshd-sftp/src/main/java/org/apache/sshd/sftp/server/AbstractSftpSubsystemHelper.java

```diff
@@ -1250,6 +1250,9 @@ protected void doRename(int id, String oldPath, String newPath, Collection<CopyO
             SftpFileSystemAccessor accessor = getFileSystemAccessor();
             accessor.renameFile(this, o, n, opts);
         } catch (IOException | RuntimeException | Error e) {
+            if (log.isDebugEnabled()) {
+                log.debug("doRename({})[id={}] Exception", getServerSession(), id, e);
+            }
```

Comment:
Hmmm. Log-then-throw is in most cases an anti-pattern. It seems that it would not be necessary to log here: just throw and let it be handled in the sendStatus() call in doRename(Buffer, int) or doPosixRename(Buffer, int).

Except that here Error is caught, too (another anti-pattern), and then re-thrown. I would expect there to be some exception quite high up in the call hierarchy logging an Error, though. So in summary I don't think this extra logging here is necessary.

Reply:
done